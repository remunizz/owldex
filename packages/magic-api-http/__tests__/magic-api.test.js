jest.mock("axios");

const { fetchCards } = require("../cjs");
const axios = require("axios");

describe("Magic service API", async () => {
  const mockCard = { name: "mock-card-name" };
  const count = 100;
  const link = "mock-link";
  const pageSize = 100;
  const rateLimit = 1000;
  const rateLimitRemaining = 999;
  const totalCount = 283;

  describe("common response", () => {
    beforeAll(() => {
      const headers = {
        count: count.toString(),
        link: link,
        "page-size": pageSize.toString(),
        "ratelimit-limit": rateLimit.toString(),
        "ratelimit-remaining": rateLimitRemaining.toString(),
        "total-count": totalCount.toString()
      };

      const mockCards = { cards: [{ ...mockCard }] };
      const mockResponse = { data: mockCards, headers: { ...headers } };
      axios.request.mockResolvedValue(mockResponse);
    });

    afterAll(() => {
      axios.request.mockReset();
    });

    it("should have a valid pagination", async () => {
      const response = await fetchCards();
      expect(response.pagination).toMatchObject({
        count,
        pageSize,
        ratelimitLimit: rateLimit,
        ratelimitRemaining: rateLimitRemaining,
        totalCount
      });
    });
  });

  describe("simple response", async () => {
    beforeAll(() => {
      const mockCards = { cards: [{ ...mockCard }] };
      const mockResponse = { data: mockCards };
      axios.request.mockResolvedValue(mockResponse);
    });

    afterAll(() => {
      axios.request.mockReset();
    });

    it("should contain mock card", async () => {
      const response = await fetchCards();
      expect(response.items.cards).toContainEqual({ ...mockCard });
    });

    it("should use queries", async () => {
      const response = await fetchCards({ name: "mock" });
      expect(response.items.cards.length).toBe(1);
    });
  });
});
