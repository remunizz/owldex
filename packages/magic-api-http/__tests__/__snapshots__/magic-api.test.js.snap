// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Magic service API common response should match snapshot 1`] = `
Object {
  "items": Object {
    "cards": Array [
      Object {
        "name": "mock-card-name",
      },
    ],
  },
  "pagination": Object {
    "count": 100,
    "link": "mock-link",
    "pageSize": 100,
    "ratelimitLimit": 1000,
    "ratelimitRemaining": 999,
    "totalCount": 283,
  },
}
`;
